samples/subsys/debug/gdbstub/pytest/test_gdbstub.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2023 Intel Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pytest
+import subprocess
+from twister_harness import DeviceAdapter
+from pathlib import Path
+import sys
+
+zbase = os.getenv("ZEPHYR_BASE")
+print(zbase)
+sys.path.insert(0, os.path.join(zbase, "scripts", "pylib", "twister"))
+from twisterlib.cmakecache import CMakeCache
+
+@pytest.fixture()
+def cmake_cache(request):
+    build_dir = Path(request.config.getoption('--build-dir'))
+    cache_file_name = build_dir / 'CMakeCache.txt'
+    return CMakeCache.from_file(cache_file_name)
+
+
+def test_gdbstub(dut: DeviceAdapter, request, cmake_cache):
```

Comment:
do we need a `cmake_cache` fixture? maybe better to just call here a method to create cache object:
`cmake_cache = CMakeCache.from_file(build_dir / 'CMakeCache.txt')`
Or if we plan to add more subtests, or want to have separated setup env from execution, than good to leave that fixture.

Reply:
probably not and can be done adhoc right now, but, it would be nice to make all of this available to pytest in some generic way, i.e. cmake cache, kconfigs (https://github.com/zephyrproject-rtos/zephyr/pull/62517/files#diff-68e2d9beb36d051a52c28ade2164f500a89902d0cd752fb79a95ec2405d3ce0dR17) and other variables that can be used for testing.

Reply:
ok, thanks.